File: mandel_test.hs
{-# LANGUAGE BangPatterns #-}

import Data.Complex

mandel :: Int -> Complex Double -> Int
mandel max_depth c = loop 0 0
  where   
   loop i !z
     | i == max_depth     = i
     | magnitude z >= 2.0 = i
     | otherwise          = loop (i+1) (z*z + c)

main = print$ mandel (5 * 1000 * 1000) (0.1 :+ 0.1)

File: mandel_test.ml
open Complex

let mandel depth c = 
  let rec loop i z = 
     if i == depth         then i 
     else if norm z >= 2.0 then i
     else loop (i+1) (add (mul z z) c)
  in loop 0 {re=0.0; im=0.0};;

print_int (mandel (5 * 1000 * 1000) {re=0.1; im=0.1});;
print_string "\n";;
